src/cli/commands/why.js

```diff
@@ -135,10 +135,18 @@ export async function run(config: Config, reporter: Reporter, flags: Object, arg
 
   const query = await cleanQuery(config, args[0]);
 
-  reporter.step(1, 4, reporter.lang('whyStart', args[0]), emoji.get('thinking_face'));
+  // Be default there are 3 steps to this task
+  let total = 3;
+
+  // If the user wants size information, we enable the fourth step
+  if (flags.size) {
```

Comment:
I think instead of doing this, you can have an array: `const steps = [];`

Start pushing the steps as functions to that:
```js
steps.push({msg: reporter.lang('whyStart', args[0]), emoji.get('thinking_face'), action: () => {}});
steps.push({msg: reporter.lang('whyInitGraph'), emoji.get('truck'), action: () => {
  const lockfile = await Lockfile.fromDirectory(config.lockfileFolder, reporter);
  const install = new Install(flags, config, reporter, lockfile);
  ...
}});

...

if (flags.size) {
  steps.push(...);
}

for (const i = 0; i < steps.length; i++) {
  const {msg, action} = steps[i];
  reporter.step(i+1, steps.length, msg);
  action();
}
```